### ISSUE TITLE
`float_tensor / python_scalar` on CUDA is not correctly rounded (reciprocal-multiply), diverging from CPU and from `tensor / tensor`

### ISSUE THREAD

**Issue by IlyasMoutawwakil** [pytorch/pytorch#191495]
### 🐛 Describe the bug

Dividing a floating-point CUDA tensor by a **Python scalar** is computed as a multiply-by-reciprocal
(`tensor * (1 / scalar)`) rather than a true IEEE-754 division. This introduces up to 1 ULP of error
**even when the exact quotient is representable**. The same operation on CPU, and `tensor / tensor`
division on CUDA, are both correctly rounded.

### Minimal reproducer

```python
import torch

a = torch.tensor(169.0)   # 169 = 13 * 13; 13.0 is exactly representable in float32

print((a.cpu()  / 13.0).item())                              # 13.0                (correct)
print((a.cuda() / 13.0).item())                              # 13.000000953674316  (1 ULP high)
print((a.cuda() * (1.0 / 13.0)).item())                      # 13.000000953674316  (== the "/ scalar" result)
print((a.cuda() / torch.tensor(13.0, device="cuda")).item()) # 13.0                (tensor/tensor is correct)
```

`169 / 13 == 13` exactly and `13.0` is exactly representable in float32, so IEEE-754 round-to-nearest
division **must** return `13.0`. CPU does. CUDA returns `13.0 + 1 ULP`, and that value is bit-identical
to `a * (1/13.0)` — i.e. the scalar-divisor path is a reciprocal multiply.

### Systematic check

Over every exactly-divisible float32 pair `a = q * b` for `2 <= b, q < 200` (39 204 cases, each with an
exactly-representable quotient `q`):

| path | results that are **not** correctly rounded |
|---|---|
| CPU  `tensor / scalar` | 0 / 39 204 |
| CUDA `tensor / tensor` | 0 / 39 204 |
| **CUDA `tensor / scalar`** | **3 807 / 39 204** |
| CUDA `tensor / scalar` cases that exactly equal `tensor * (1/scalar)` | **3 807 / 3 807** |

A few CUDA `tensor / scalar` failures:

```
21.0 / 7  ->  3.000000238418579   (exact 3.0)
49.0 / 7  ->  7.000000476837158   (exact 7.0)
91.0 / 7  -> 13.000000953674316   (exact 13.0)
```

## Expected behavior

`tensor / scalar` should be IEEE-754 correctly-rounded (round-to-nearest-even), matching both CPU
`tensor / scalar` and CUDA `tensor / tensor`. When the exact quotient is representable it should be
returned exactly.

## Root cause

The CUDA scalar-divisor path folds the divisor into a host-computed reciprocal and multiplies
(`tensor * (1 / scalar)`). Reciprocal-multiply is not correctly rounded. `tensor / tensor` uses a true
division kernel (`div.rn.f32`) and is correct; `tensor / scalar` does not. Every one of the 3 807
failing cases above is bit-identical to `tensor * (1 / scalar)`, confirming the mechanism.

### Versions

Reproduces identically on:

- `2.12.0+cu126`
- `2.13.0+cu126` (latest stable)
- `2.14.0.dev20260729+cu126` (nightly / `main`)

Environment: CUDA 12.6, NVIDIA A100-SXM4-80GB.

cc @ptrblck @msaroufim @eqy @tinglvv @nWEIdia

**Comment by MDSALMANSHAMS** [comment]
Confirmed the root cause in source: it's the deliberate CPU-scalar fast path in `div_true_kernel_cuda` (`aten/src/ATen/native/cuda/BinaryDivTrueKernel.cu`). When the divisor is a CPU scalar (`iter.is_cpu_scalar(2)`), the kernel computes `inv_b = high_prec_t(1.0) / scalar` on the host and then **multiplies** (`a * inv_b`) rather than dividing -- there's even an in-code comment on that branch noting it "may lose one bit". That multiply-by-reciprocal is exactly the up-to-1-ULP divergence from CPU `tensor / scalar` and from CUDA `tensor / tensor`, both of which go through the true `DivFunctor`.

So this is a known perf-vs-accuracy tradeoff baked into that branch, which makes it a policy call rather than a plain bug: e.g. route the floating-point scalar-divisor case through the true `DivFunctor` (correctly rounded, matching CPU and tensor/tensor), or keep the reciprocal-multiply only behind an opt-in fast path.

Would the team accept making `tensor / scalar` correctly-rounded here? Happy to put up the change plus a ULP-parity test -- could a maintainer confirm the intended behavior and mark this `actionable`?

**Comment by malfet** [comment]
PyTorch does not guarantee bitwise-identical floating-point results across devices or between different operator implementations. The CUDA scalar-divisor path intentionally uses multiplication by a precomputed reciprocal; as the source comment notes, this optimization may lose one bit of precision.
The reported results remain within the expected numerical accuracy for this operation, so the difference from CPU and CUDA tensor/tensor division is not considered a correctness bug. Closing as not planned.
